src/BlockInfo.h

```diff
@@ -79,6 +82,10 @@ class cBlockInfo
 	inline static bool IsTransparent              (BLOCKTYPE a_Type) { return Get(a_Type).m_Transparent;         }
 	inline static bool IsOneHitDig                (BLOCKTYPE a_Type) { return Get(a_Type).m_OneHitDig;           }
 	inline static bool IsPistonBreakable          (BLOCKTYPE a_Type) { return Get(a_Type).m_PistonBreakable;     }
+	inline static bool IsRainBlocker              (BLOCKTYPE a_Type)
+	{
+		return ((Get(a_Type).m_IsRainBlocker) || Get(a_Type).m_IsSolid);
```

Comment:
One thought I had was to iterate over all the block infos (in cBlockInfoArray()) and manually set IsRainBlocker for every block that has IsSolid. That way in the future there could be solid blocks that don't block rain (and, more immediately, this method could just return `Get(a_Type).m_IsRainBlocker`).

Regardless, `Get(a_Type).m_IsRainBlocker` doesn't need parentheses around it.

Reply:
I would go a step further and question the whole `cBlockInfo` class. Why not just a set of simple functions?

But for this PR I agree with @lkolbly : Make `IsRainBlocker` and `m_IsRainBlocker` consistent or make `Get` and all `cBlockInfo` variables private, so nobody can accidentially write `Get(mySpecialBlock).m_IsRainBlocker`.